pandas/tseries/period.py

```diff
@@ -211,79 +247,16 @@ def _generate_range(cls, start, end, periods, freq, fields):
         return subarr, freq
 
     @classmethod
-    def _from_arraylike(cls, data, freq, tz):
-        if not isinstance(data, (np.ndarray, PeriodIndex,
-                                 DatetimeIndex, Int64Index)):
-            if lib.isscalar(data) or isinstance(data, Period):
-                raise ValueError('PeriodIndex() must be called with a '
-                                 'collection of some kind, %s was passed'
-                                 % repr(data))
+    def _simple_new(cls, data, name=None, freq=None, **kwargs):
```

Comment:
Had a quick go at simplifying it - this passes and takes out all of the custom stuff except for back-compat + a one line `__getstate__`. But also agree, not a big deal
https://github.com/SixtyCapital/pandas/compare/period-float...SixtyCapital:use-getstate